spec/datadog/profiling/collectors/discrete_dynamic_sampler_spec.rb

```diff
@@ -0,0 +1,226 @@
+require 'datadog/profiling/spec_helper'
+require 'datadog/profiling'
+
+RSpec.describe 'Datadog::Profiling::Collectors::DiscreteDynamicSampler' do
+  let(:max_overhead_target) { 2.0 }
+
+  before do
+    skip_if_profiling_not_supported(self)
+    @now = Time.now.to_f
+  end
+
+  subject(:sampler) do
+    sampler = Datadog::Profiling::Collectors::DiscreteDynamicSampler::Testing::Sampler.new
+    update_overhead_target(max_overhead_target, sampler)
+    sampler
+  end
+
+  def maybe_sample(sampling_seconds:)
+    start_ns = (@now * 1e9).to_i
+    end_ns = start_ns + (sampling_seconds * 1e9).to_i
+    sampler._native_after_sample(end_ns) / 1e9 if sampler._native_should_sample(start_ns)
+  end
+
+  def simulate_load(duration_seconds:, events_per_second:, sampling_seconds:)
+    start = @now
+    num_events = (events_per_second.to_f * duration_seconds).to_i
+    time_between_events = duration_seconds.to_f / num_events
+    num_samples = 0
+    total_sampling_seconds = 0
+    num_events.times do
+      sampling_time = maybe_sample(sampling_seconds: sampling_seconds)
+      unless sampling_time.nil?
+        num_samples += 1
+        total_sampling_seconds += sampling_time
+        @now += sampling_time
+      end
+      @now += time_between_events
+    end
+    {
+      sampling_ratio: num_samples.to_f / num_events,
+      total_sampling_seconds: total_sampling_seconds,
+      overhead: total_sampling_seconds / (@now - start),
+      num_samples: num_samples,
+      num_events: num_events,
+    }
+  end
+
+  def update_overhead_target(new_overhead_target, sampler_instance = sampler)
+    sampler_instance._native_set_overhead_target_percentage(new_overhead_target, @now * 1e9)
+  end
+
+  context 'when under a constant' do
+    let(:events_per_second) { nil }
```

Comment:
Minor: Since the default doesn't get used (and would break anyway?)

```suggestion
  context 'when under a constant' do
```